pkg/xray/resource_xray_vulnerabilities_report.go

```diff
@@ -0,0 +1,135 @@
+package xray
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+func resourceXrayVulnerabilitiesReport() *schema.Resource {
+	var vulnerabilitiesFilterSchema = map[string]*schema.Schema{
+		"vulnerable_component": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "Filter by component name, you can use (*) at the beginning or end of a substring as a wildcard.",
+		},
+		"impacted_artifact": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "Filter by artifact name, you can use (*) at the eginning or end of a substring as a wildcard.",
+		},
+		"has_remediation": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Whether the issue has a fix or not.", // UI has an option 'All', when the field is empty. Not clear how to make it work with bool.
+		},
+		"cve": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "CVE.",
+		},
+		"issue_id": {
+			Type:             schema.TypeString,
+			Optional:         true,
+			ValidateDiagFunc: validator.StringIsNotEmpty,
+			Description:      "Issue ID.",
+		},
+		"severities": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Set:         schema.HashString,
+			Optional:    true,
+			Description: "Severity levels. Allowed values: 'Low', 'Medium', 'High', 'Critical'",
```

Comment:
Please add validation func for the allowed values.

Reply:
Yes, missed that.

Reply:
Oh, actually I haven't missed it. 
```
resource xray_vulnerabilities_report: severities: ValidateFunc and 
ValidateDiagFunc are not yet supported on lists or sets.
```

Reply:
Right, then we should add validation logic to the customizeDiff func

Reply:
Added